src/timing/timer/tests/mod.rs

```diff
@@ -617,3 +617,51 @@ fn reset_and_set_attempt_as_pb() {
     assert_eq!(attempt.time().game_time, None);
     assert!(attempt.ended().unwrap().time >= attempt.started().unwrap().time);
 }
+
+#[test]
+fn fix_run_upon_creation() {
+    let mut timer = create_timer(&["A", "B"]);
+    run_with_splits(&mut timer, &[3.0, 6.0]);
+    run_with_splits(&mut timer, &[2.0, 4.0]);
+    let mut run = timer.into_run(true);
+
+    // We pop the last attempt from the history, but keep it in the segment
+    // history, which makes the segment history elements unattached.
+    run.attempt_history.pop().unwrap();
+
+    // This should cause the run to be fixed.
+    let run = Timer::new(run).unwrap().into_run(true);
+
+    let segments = run.segments();
+
+    assert_eq!(segments[0].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[0].segment_history().try_get_max_index(), Some(1));
+
+    assert_eq!(segments[1].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[1].segment_history().try_get_max_index(), Some(1));
+}
+
+#[test]
+fn fix_run_upon_replacement() {
+    let mut timer = create_timer(&["A", "B"]);
+    run_with_splits(&mut timer, &[3.0, 6.0]);
+    run_with_splits(&mut timer, &[2.0, 4.0]);
+    let mut run = timer.into_run(true);
+
+    // We pop the last attempt from the history, but keep it in the segment
+    // history, which makes the segment history elements unattached.
+    run.attempt_history.pop().unwrap();
+
+    // This should cause the run to be fixed.
+    let mut timer = create_timer(&["A", "B"]);
+    timer.set_run(run);
+    let run = timer.into_run(true);
+
+    let segments = run.segments();
+
+    assert_eq!(segments[0].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[0].segment_history().try_get_max_index(), Some(1));
+
+    assert_eq!(segments[1].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[1].segment_history().try_get_max_index(), Some(1));
+}
```

Comment:
These tests will not compile due to run.attempt_history being private.

Should I move these tests to `crate::run::tests`, or is there something else I can do here?

Reply:
https://docs.rs/livesplit-core/0.11.0/livesplit_core/run/struct.Run.html#method.attempt_history

Reply:
Oh wait, you try to modify it. Uhm, then yeah I guess move it there